In [1]:
# write your code below in RStudio
# then press the Source button to run it

library(tidyverse)



Registered S3 methods overwritten by 'ggplot2':
  method         from 
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
Registered S3 method overwritten by 'rvest':
  method            from
  read_xml.response xml2
-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v ggplot2 3.1.1       v purrr   0.3.2  
v tibble  2.1.1       v dplyr   0.8.0.1
v tidyr   0.8.3       v stringr 1.4.0  
v readr   1.3.1       v forcats 0.4.0  
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


In [3]:
d <- read.csv("all-ages.csv")


In [4]:
print(d)


    Major_code
1         1100
2         1101
3         1102
4         1103
5         1104
6         1105
7         1106
8         1199
9         1301
10        1302
11        1303
12        1401
13        1501
14        1901
15        1902
16        1903
17        1904
18        2001
19        2100
20        2101
21        2102
22        2105
23        2106
24        2107
25        2201
26        2300
27        2301
28        2303
29        2304
30        2305
31        2306
32        2307
33        2308
34        2309
35        2310
36        2311
37        2312
38        2313
39        2314
40        2399
41        2400
42        2401
43        2402
44        2403
45        2404
46        2405
47        2406
48        2407
49        2408
50        2409
51        2410
52        2411
53        2412
54        2413
55        2414
56        2415
57        2416
58        2417
59        2418
60        2419
61        2499
62        2500
63        2501
64        2502
65        2503
66        

In [5]:
View(d)


ERROR: Error in View(d): 'View()' not yet supported in the Jupyter R kernel


The goal of this lab is to determine if the median salaries of those in the Major_category of Computers & Mathematics are significantly different than the median salaries of those in the Major_category of Education. Our hypotheses are therefore:

First, you’ll want to extract the most relevant information from the data which is data related to those in the Major_category of Computers & Mathematics

In [6]:
df1 <- filter(d, Major_category == "Computers & Mathematics")

In [7]:
print(df1)

   Major_code                                           Major
1        2001                      COMMUNICATION TECHNOLOGIES
2        2100                COMPUTER AND INFORMATION SYSTEMS
3        2101        COMPUTER PROGRAMMING AND DATA PROCESSING
4        2102                                COMPUTER SCIENCE
5        2105                            INFORMATION SCIENCES
6        2106 COMPUTER ADMINISTRATION MANAGEMENT AND SECURITY
7        2107      COMPUTER NETWORKING AND TELECOMMUNICATIONS
8        3700                                     MATHEMATICS
9        3701                             APPLIED MATHEMATICS
10       3702                 STATISTICS AND DECISION SCIENCE
11       4005                MATHEMATICS AND COMPUTER SCIENCE
            Major_category  Total Employed Employed_full_time_year_round
1  Computers & Mathematics  62141    49609                         37261
2  Computers & Mathematics 253782   218248                        189950
3  Computers & Mathematics  29317    

Next, since you are interested in the median salaries, you want to extract only the data from the Median column. To do so, you can use the c() function to extract the data in Median and turn it into a vector.

In [8]:
sal1 <- c(df1$Median)

In [9]:
print(sal1)

 [1] 50000 65000 60000 78000 68000 55000 55000 66000 70000 70000 92000


Follow the directions in Step 1 to extract only the data from d where Major_category is Education using the filter() function.

In [10]:
df2 <- filter(d, Major_category == "Education")
print(df2)

   Major_code                                       Major Major_category
1        2300                           GENERAL EDUCATION      Education
2        2301  EDUCATIONAL ADMINISTRATION AND SUPERVISION      Education
3        2303                   SCHOOL STUDENT COUNSELING      Education
4        2304                        ELEMENTARY EDUCATION      Education
5        2305               MATHEMATICS TEACHER EDUCATION      Education
6        2306      PHYSICAL AND HEALTH EDUCATION TEACHING      Education
7        2307                   EARLY CHILDHOOD EDUCATION      Education
8        2308      SCIENCE AND COMPUTER TEACHER EDUCATION      Education
9        2309                 SECONDARY TEACHER EDUCATION      Education
10       2310                     SPECIAL NEEDS EDUCATION      Education
11       2311 SOCIAL SCIENCE OR HISTORY TEACHER EDUCATION      Education
12       2312          TEACHER EDUCATION: MULTIPLE LEVELS      Education
13       2313                LANGUAGE AND DRAMA EDU

Follow the direction in Step 2 to convert the Median column of data from the Major_category of Education into a vector using the c() function.

In [11]:
sal2 <- c(df2$Median)

In [12]:
print(sal2)

 [1] 43000 58000 41000 40000 43000 48400 35300 46000 45000 42000 45000 40000
[13] 42000 42600 50000 40000


Now that you have the two numeric vectors of the median salaries of Computers & Mathematics and Education, you can use them for data analysis.

In [14]:
t.test(sal1, sal2, paired = FALSE, var.equal = TRUE)


	Two Sample t-test

data:  sal1 and sal2
t = 6.768, df = 25, p-value = 4.297e-07
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 15612.39 29270.56
sample estimates:
mean of x mean of y 
 66272.73  43831.25 


Enter the EXACT value of the produced mean of the median salaries of the major category Computers & Mathematics that you generated from your analysis. DO NOT ROUND!

If you performed the analysis correctly, the expected mean for the median salaries of Computers & Mathematics is: 66272.73

Enter the EXACT value of the produced mean of the median salaries of the major category Education that you generated from your analysis. DO NOT ROUND!

If you performed the analysis correctly, the expected mean for the median salaries of Education is: 43831.25

Based on the generated p-value, SELECT ALL of the following that are TRUE.

The p-value is statistically significant.

We reject the null hypothesis in favor of the alternative hypothesis.

The results show that there is a significant difference between the median salaries of Computers & Mathematics majors versus Education majors.